pkg/oauth/server.go

```diff
@@ -245,20 +254,22 @@ func (s *server) RegisterRoutes(server *web.Server) {
 
 	api := group.Group("/api", middleware.CSRF())
 	api.POST("/auth/login", s.Login)
-	api.POST("/auth/logout", s.Logout, s.requireLogin)
 	api.GET("/me", s.CurrentUser, s.requireLogin)
 
 	page := group.Group("", middleware.CSRFWithConfig(middleware.CSRFConfig{
 		TokenLookup: "form:csrf",
 	}))
 	page.GET("/login", webui.Template.Handler, s.redirectToNext)
+	page.GET("/logout", s.Logout, s.requireLogin)
```

Comment:
We may need to replace `requireLogin` with something else, otherwise logout+redirect will fail if the user is already logged out from the OAuth provider.